lib/prometheus_exporter/server/collector.rb

```diff
@@ -1,8 +1,13 @@
 # frozen_string_literal: true
 
+require "logger"
+
 module PrometheusExporter::Server
   class Collector < CollectorBase
-    def initialize(json_serializer: nil)
+    attr_reader :logger
+
+    def initialize(json_serializer: nil, logger: Logger.new(STDERR))
```

Comment:
Added a defaulted logger targeted to STDERR to match what behavior was here before, can also change this. By default it's passing in the server logger though in the main flows.